src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs

```diff
@@ -172,16 +176,18 @@ public virtual void Close()
             {
                 this.initialized = false;
 
-                // We don't need to terminate if the test host has already terminated. The upper bound
-                // for wait should be 100ms.
-                if (this.testHostProcessId != -1 && !this.testHostExited.Wait(100))
+                // We don't need to terminate if the test host has already terminated.
+                // If the host launch fails, please clean test host. 
```

Comment:
Can the pattern be simple?
* If the consumer is exiting, always call `CleanTestHostAsync`. It is the responsibility of `Clean` to ensure that there are no lingering child processes

If there were no test host, it is up to `CleanTestHostAsync` to do what it chooses.

Thinking aloud, does `IDisposable` or `Close` make more sense than `CleanXyz`?